File: week1/transpose-matrix.java
class Solution {
    public int[][] transpose(int[][] A) {
        int[][] rows = new int[A[0].length][A.length];
        for (int row = 0; row < A.length; row++) {
            for (int col = 0; col < A[0].length; col++) {
                rows[col][row] = A[row][col];
            }
        }
        return rows;
    }
}

File: 2026/week1/transpose-matrix.py
class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        rows = len(matrix)
        cols = len(matrix[0])
        result = [[0] * rows for _ in range(cols)]
        for row in range(rows):
            for col in range(cols):
                result[col][row] = matrix[row][col]
        return result
